includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php

```diff
@@ -0,0 +1,1000 @@
+	 * @return WP_Error|bool
+	 */
+	public function get_items_permissions_check( $request ) {
+		$context = $request->get_param( 'context' );
+		$retval  = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not allowed to see the notices.', 'buddypress' ),
+			array(
+				'status' => rest_authorization_required_code(),
+			)
+		);
+
+		if ( is_user_logged_in() ) {
+			if ( 'view' === $context ) {
+				$retval = true;
+			} elseif ( 'edit' === $context ) {
+				$retval = bp_current_user_can( 'bp_moderate' );
+			}
+		}
+
+		/**
+		 * Filter the messages `get_items` permissions check.
+		 *
+		 * @since 9.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_rest_sitewide_notices_get_items_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Get a single notice by ID.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$notice = $this->get_notice_object( $request->get_param( 'id' ) );
+
+		$retval   = $this->prepare_item_for_response( $notice, $request );
+		$response = rest_ensure_response( $retval );
+
+		/**
+		 * Fires after a sitewide notice is fetched via the REST API.
+		 *
+		 * @since 9.0.0
+		 *
+		 * @param BP_Messages_Notice $notice  Notice object.
+		 * @param WP_REST_Response   $retval  The response data.
+		 * @param WP_REST_Request    $request The request sent to the API.
+		 */
+		do_action( 'bp_rest_sitewide_notices_get_item', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request is allowed to get a sitewide notice.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|bool
+	 */
+	public function get_item_permissions_check( $request ) {
+
+		$retval = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not allowed to read notices.', 'buddypress' ),
+			array(
+				'status' => rest_authorization_required_code(),
+			)
+		);
+
+		if ( is_user_logged_in() ) {
+			$notice = $this->get_notice_object( $request->get_param( 'id' ) );
+			if ( empty( $notice->id ) ) {
+				$retval = new WP_Error(
+					'bp_rest_invalid_id',
+					__( 'Sorry, this notice does not exist.', 'buddypress' ),
+					array(
+						'status' => 404,
+					)
+				);
+			} else {
+				if ( bp_current_user_can( 'bp_moderate' ) ) {
+					$retval = true;
+				} else {
+					// Non-admin users can only see the active notice.
+					$is_active = isset( $notice->is_active ) ? $notice->is_active : false;
+					if ( ! $is_active ) {
+						$retval = new WP_Error(
+							'bp_rest_authorization_required',
+							__( 'Sorry, you are not allowed to see this notice.', 'buddypress' ),
+							array(
+								'status' => rest_authorization_required_code(),
+							)
+						);
```

Comment:
Why not use the same error set in the beginning of this method? They are almost the same.
```suggestion
						$retval = $retval;
```

Reply:
I agree! Thanks for your suggestion.